### PR TITLE
RFC: remove deprecation about __precompile__(true)

### PR DESCRIPTION
**Pull request by KristofferC**
Doesn't seem to be much value in deprecating this?

### CODE REVIEW

File: base/loading.jl [hunk 1]
```diff
@@ -769,16 +769,14 @@ precompilableerror(@nospecialize ex) = false
...
-Specify that the file calling this function is not precompilable.
+Specify that the file calling this function is not precompilable, defaulting to `true`.
 If a module or file is *not* safely precompilable, it should call `__precompile__(false)` in
 order to throw an error if Julia attempts to precompile it.
 """
 @noinline function __precompile__(isprecompilable::Bool=true)
-    if isprecompilable
-        depwarn("__precompile__() is now the default", :__precompile__)
-    elseif 0 != ccall(:jl_generating_output, Cint, ())
+    if ccall(:jl_generating_output, Cint, ()) != 0
```
Comment (JeffBezanson) [inline base/loading.jl]: `&& !isprecompilable` ?

File: base/loading.jl [hunk 2]
```diff
@@ -769,16 +769,14 @@ precompilableerror(@nospecialize ex) = false
 
 # Call __precompile__(false) at the top of a tile prevent it from being precompiled (false)
 """
-    __precompile__(false)
+    __precompile__(isprecompilable::Bool)
 
-Specify that the file calling this function is not precompilable.
+Specify that the file calling this function is not precompilable, defaulting to `true`.
```
Comment (ararslan) [inline base/loading.jl]: The wording makes it sound like precompilation is _disabled_ by default. I'd reword as

> Specify whether the file calling this function is precompilable, defaulting to `true`. If this function is not called explicitly, precompilation will be enabled by default.

**Comment by KristofferC** [comment]
Updated.

**Review by ararslan — APPROVED** [review]
It'd be nice to mention in the docstring that not using the function at all defaults to precompiling, but that doesn't have to happen here.

**Comment by KristofferC** [comment]
Yeah, that's a good point.

**Comment by ararslan** [comment]
Thinking about this more, isn't the deprecation message fine to have for 0.7? It will tell people what the behavior will be in 1.0 before we silently break it.

**Comment by JeffBezanson** [comment]
That would be a different message though --- in the case where there's no `__precompile__` call, not where there is a `__precompile__()` call.

**Comment by ararslan** [comment]
I see. It seems definitely worthwhile to me to emit a warning when there's no `__precompile__` call at all.

**Comment by KristofferC** [comment]
Separate PR? :)

**Comment by ararslan** [comment]
Yeah definitely

**Comment by Keno** [comment]
> I see. It seems definitely worthwhile to me to emit a warning when there's no __precompile__ call at all.

I don't think we should do this. It's perfectly legal to have no __precompile__ call now and the previous depwarn has caused people to remove theirs.

**Comment by KristofferC** [comment]
> the previous depwarn has caused people to remove theirs.

Yeah, the depreciation should have deprecated having no precompile statement forcing people to do `__precompile__(false)` while leaving `__precompile__(true)` alone. Forcing the people that removed their precompile statement to add them back seems quite cruel though.

**Comment by JeffBezanson** [comment]
Agreed. I also think most/all packages that don't support precompilation have already marked `__precompile__(false)`.
